### PR TITLE
feat(wallet-ui): layout

### PR DESCRIPTION
**Pull request by gregfromstl**
## Problem solved

Short description of the bug fixed or feature added

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [ ] Internal API changes: explain the internal logic changes

## How to test

- [ ] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

## Contributor NFT

Paste in your wallet address below and we will airdrop you a special NFT when your pull request is merged. 

```Address: ```

---

## PR-Codex overview
This PR updates various components, adds third-party dependencies, integrates Tailwind CSS, and includes ecosystem API calls.

### Detailed summary
- Added `Erc20Token` and `Erc721Token` types
- Updated client creation in `client.ts`
- Implemented `cn`, `Skeleton`, `Card`, `ConnectButton`, `Toaster`, `Label`, `Input`, and `Badge` components
- Modified layout in `layout.tsx` and `page.tsx`
- Included `getEcosystemInfo` and `getChains` functions
- Integrated third-party dependencies like `@radix-ui/react-dialog`, `@radix-ui/react-label`, `@radix-ui/react-popover`, and more

> The following files were skipped due to too many changes: `apps/wallet-ui/src/components/ui/badge.tsx`, `apps/wallet-ui/src/lib/assets/erc20.ts`, `apps/wallet-ui/src/components/ui/popover.tsx`, `apps/wallet-ui/src/app/[ecosystem]/(ui)/wallet/[address]/layout.tsx`, `apps/wallet-ui/src/app/[ecosystem]/(ui)/wallet/[address]/page.tsx`, `apps/wallet-ui/src/app/[ecosystem]/(ui)/layout.tsx`, `apps/wallet-ui/src/components/ui/button.tsx`, `apps/wallet-ui/src/components/NftGallery.tsx`, `apps/wallet-ui/src/lib/assets/erc721.ts`, `apps/wallet-ui/src/components/NftCard.tsx`, `apps/wallet-ui/src/util/simplehash.ts`, `apps/wallet-ui/src/components/ChainCombobox.tsx`, `apps/wallet-ui/src/components/TransactionToastProvider.tsx`, `apps/wallet-ui/src/components/ui/use-toast.ts`, `apps/wallet-ui/src/app/globals.css`, `apps/wallet-ui/src/components/ui/dialog.tsx`, `apps/wallet-ui/src/components/ChainIcon.tsx`, `apps/wallet-ui/src/components/ui/form.tsx`, `apps/wallet-ui/src/components/NftModal.tsx`, `apps/wallet-ui/src/components/ui/toast.tsx`, `apps/wallet-ui/src/components/ui/command.tsx`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3511?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3511** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3511?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3498** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3498?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3499](https://github.com/thirdweb-dev/js/pull/3499) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3499?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3478** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3478?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3477** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3477?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/wallet-ui/biome.json [hunk 1]
```diff
@@ -7,6 +7,6 @@
     }
   },
   "files": {
-    "ignore": []
+    "ignore": ["./src/components/ui/**/*"]
```
Comment (jnsdls) [inline apps/wallet-ui/biome.json:10]: why?

File: apps/wallet-ui/next.config.mjs [hunk 2]
```diff
@@ -1,4 +1,14 @@
 /** @type {import('next').NextConfig} */
-const nextConfig = {};
+const nextConfig = {
+  images: {
+    dangerouslyAllowSVG: true,
+    remotePatterns: [
+      {
+        protocol: "https",
+        hostname: "**",
```
Comment (jnsdls) [inline apps/wallet-ui/next.config.mjs:8]: this feels potentially dangerous, it means anyone can use this project as a free (not for us!) image resizing service potentially, no?

File: apps/wallet-ui/package.json [hunk 3]
```diff
@@ -10,19 +10,35 @@
...
+    "tiny-invariant": "^1.3.3",
+    "zod": "^3.23.8"
   },
   "devDependencies": {
     "@types/node": "^20",
     "@types/react": "^18",
     "@types/react-dom": "^18",
     "eslint": "^8",
-    "eslint-config-next": "14.2.4",
+    "eslint-config-next": "15.0.0-rc.0",
```
Comment (jnsdls) [inline apps/wallet-ui/package.json:41]: downgrade back to 14 please

File: apps/wallet-ui/src/app/globals.css [hunk 4]
```diff
@@ -1,70 +1,85 @@
 @tailwind base;
```
Comment (jnsdls) [inline apps/wallet-ui/src/app/globals.css:1]: I'd assume most of this should be 1:1 the same as dashboard?

File: apps/wallet-ui/src/components/ui/toast.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
+"use client";
```
Comment (jnsdls) [inline apps/wallet-ui/src/components/ui/toast.tsx:1]: the [sonner](https://ui.shadcn.com/docs/components/sonner) stuff feels superior to this imo

File: apps/wallet-ui/src/lib/ecosystems.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+export async function getEcosystemInfo(ecosystemId: string) {
+  const response = await fetch(
+    "https://embedded-wallet.thirdweb.com/api/2024-05-05/ecosystem-wallet",
+    {
+      headers: {
+        "x-ecosystem-id": "ecosystem.new.age", // Hardcoded for now
```
Comment (jnsdls) [inline apps/wallet-ui/src/lib/ecosystems.ts:6]: obv can't be in prod

File: apps/wallet-ui/src/lib/revalidate.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+"use server";
+
+import { revalidatePath, revalidateTag } from "next/cache";
+
+export async function revalidate(tag: string) {
+  revalidateTag(tag);
```
Comment (jnsdls) [inline apps/wallet-ui/src/lib/revalidate.ts:6]: what are we revalidating?

File: apps/wallet-ui/tailwind.config.ts [hunk 8]
```diff
@@ -1,9 +1,13 @@
...
-/** @type {import('tailwindcss').Config} */
-module.exports = {
+const config = {
   darkMode: ["class"],
-  content: ["./src/**/*.{ts,tsx}"],
+  content: [
+    './pages/**/*.{ts,tsx}',
+    './components/**/*.{ts,tsx}',
+    './app/**/*.{ts,tsx}',
+    './src/**/*.{ts,tsx}',
```
Comment (jnsdls) [inline apps/wallet-ui/tailwind.config.ts:9]: why did this have to become more explicit?

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/wallet-ui/tailwind.config.ts (hunk 8 above)
Comment (gregfromstl) [reply apps/wallet-ui/tailwind.config.ts:9]: shad generated this I think

File: apps/wallet-ui/src/lib/ecosystems.ts (hunk 6 above)
Comment (gregfromstl) [reply apps/wallet-ui/src/lib/ecosystems.ts:6]: yep

File: apps/wallet-ui/src/lib/revalidate.ts (hunk 7 above)
Comment (gregfromstl) [reply apps/wallet-ui/src/lib/revalidate.ts:6]: Anything, Next just won't let me call revalidateTag directly in a client component

File: apps/wallet-ui/src/lib/ecosystems.ts (hunk 6 above)
Comment (gregfromstl) [reply apps/wallet-ui/src/lib/ecosystems.ts:6]: yep

File: apps/wallet-ui/src/app/globals.css (hunk 4 above)
Comment (gregfromstl) [reply apps/wallet-ui/src/app/globals.css:1]: Everything but the primary / secondary colors (we don't want our branding)

File: apps/wallet-ui/next.config.mjs (hunk 2 above)
Comment (gregfromstl) [reply apps/wallet-ui/next.config.mjs:12]: Is there a better way to display the image for any NFT?

File: apps/wallet-ui/biome.json (hunk 1 above)
Comment (gregfromstl) [reply apps/wallet-ui/biome.json:5]: The shad components do funky things biome doesn't like

**Pull request merged**